pkg/policy/mapstate.go

```diff
@@ -1007,6 +1068,14 @@ func (ms *mapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapState
 			}
 		}
 		if !bailed {
```

Comment:
if we bailed might as well exit early ie if bailed { return }

Reply:
This makes less difference here, but since I added a commit to that effect the main branch version (#33953), I added one here as well :-)